src/clang.rs

```diff
@@ -1789,9 +1794,15 @@ pub fn ast_dump(c: &Cursor, depth: isize) -> CXChildVisitResult {
                 format!(" {}number-of-template-args = {}", prefix, num),
             );
         }
-        if let Some(width) = c.bit_width() {
-            print_indent(depth, format!(" {}bit-width = {}", prefix, width));
-        }
+
+        // It is not safe to check the bit width without ensuring it doesn't
+        // depend on some template parameter. See
+        // https://github.com/rust-lang/rust-bindgen/issues/2239
+
+        // if let Some(width) = c.bit_width() {
+        //     print_indent(depth, format!(" {}bit-width = {}", prefix, width));
+        // }
```

Comment:
This wasn't possible because still there was no way to query if `bit_width` was callable.

I restructured the code to allow this. I had to move some logic into `clang::Cursor`: I added methods for getting a cursor to a bit field's width expression, and for checking if a cursor's AST subtree refers to a template parameter.